source/core/GenericFunctions.cpp

```diff
@@ -683,13 +683,18 @@ VIREO_FUNCTION_SIGNATURET(VectorMaxMinOp, AggregateMaxAndMinInstruction)
     IntIndex elementSizeY = srcArrayY->ElementType()->TopAQSize();
     IntIndex elementSizeMax = maxArray->ElementType()->TopAQSize();
     IntIndex elementSizeMin = minArray->ElementType()->TopAQSize();
+    IntIndex* newDimensionLengths = srcArrayX->DimensionLengths();
+    IntIndex rank = srcArrayX->Rank();
     IntIndex count = srcArrayX->Length();
-    if (srcArrayY->Length() < count)
+    if (srcArrayY->Length() < count) {
```

Comment:
I agree, this is wrong.  It actually needs to find the minimum of all the dimsizes individually and use the ArrayIterator to traverse the dims, like the VectorVectorBinaryOp does.
The max/min of a 4x2 array and a 2x4 array should produce a 2x2 array and ignore the extra elements in each array the other one doesn't have.